pkg/markdown/html_links.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package markdown
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+var (
+	htmlTagLinkRegex    = regexp.MustCompile(`<\b[^>]*?\b((?i)href|(?i)src)\s*=\s*(\"([^"]*\")|'[^']*'|([^'">\s]+))`)
+	htmlTagLinkURLRegex = regexp.MustCompile(`((http|https|ftp|mailto):\/\/)?(\.?\/?[\w\.\-]+)+\/?([#?=&])?`)
+)
+
+// OnHTMLink is a callback function invoked on each link (src|href attribute)
+// found in HTML tag.
+// It is supplied a link destination and is expected to return destination,
+// that will update the link or error.
+type OnHTMLink func(destination []byte) ([]byte, error)
```

Comment:
I understand you stick to the convention used for the other links, but what do you think about using something like ProcessDestinationLinkFunc or Compute, in order to be self explanatory? It is not necessary to be done in this PR anyways.

Reply:
Changed to reflect the intent better